hoomd/BoxResizeUpdater.cc

```diff
@@ -45,105 +39,142 @@ BoxResizeUpdater::~BoxResizeUpdater()
 
             for (unsigned int i = 0; i < m_pdata->getN(); i++)
                 {
                 // obtain scaled coordinates in the old global box
-                Scalar3 f = curBox.makeFraction(make_scalar3(h_pos.data[i].x, h_pos.data[i].y, h_pos.data[i].z));
-
-                // intentionally scale both rigid body and free particles, this may waste a few cycles but it enables
-                // the debug inBox checks to be left as is (otherwise, setRV cannot fixup rigid body positions without
-                // failing the check)
-                Scalar3 scaled_pos = newBox.makeCoordinates(f);
+                Scalar3 fractional_pos = cur_box.makeFraction(
+                    make_scalar3(h_pos.data[i].x,
+                                 h_pos.data[i].y,
+                                 h_pos.data[i].z));
+
+                // intentionally scale both rigid body and free particles, this
+                // may waste a few cycles but it enables the debug inBox checks
+                // to be left as is (otherwise, setRV cannot fixup rigid body
+                // positions without failing the check)
+                Scalar3 scaled_pos = new_box.makeCoordinates(fractional_pos);
                 h_pos.data[i].x = scaled_pos.x;
                 h_pos.data[i].y = scaled_pos.y;
                 h_pos.data[i].z = scaled_pos.z;
                 }
             }
+        // otherwise, we need to ensure that the particles are still in their
+        // local boxes by wrapping them if they are not
         else
             {
-            // otherwise, we need to ensure that the particles are still in the (local) box
-            // move the particles to be inside the new box
-            ArrayHandle<Scalar4> h_pos(m_pdata->getPositions(), access_location::host, access_mode::readwrite);
-            ArrayHandle<int3> h_image(m_pdata->getImages(), access_location::host, access_mode::readwrite);
+            ArrayHandle<Scalar4> h_pos(m_pdata->getPositions(),
+                                       access_location::host,
+                                       access_mode::readwrite);
+
+            ArrayHandle<int3> h_image(m_pdata->getImages(),
+                                      access_location::host,
+                                      access_mode::readwrite);
 
             const BoxDim& local_box = m_pdata->getBox();
 
             for (unsigned int i = 0; i < m_pdata->getN(); i++)
                 {
-                // need to update the image if we move particles from one side of the box to the other
+                // need to update the image if we move particles from one side
+                // of the box to the other
                 local_box.wrap(h_pos.data[i], h_image.data[i]);
                 }
             }
-
         }
-
     if (m_prof) m_prof->pop();
     }
 
+BoxDim getBoxDimFromPyObject(pybind11::object box)
+    {
+    auto type_name = box.get_type().attr("__name__").cast<std::string>();
```

Comment:
This check is an extra precaution the `BoxDim` class is not user facing it is in `_hoomd`. Using `ParameterDict`, it does an isinstance check when setting on the Python side. Doing an `isinstance` check here would be harder since we would have to import the Python module using pybind and then grab the `Box` class. I don't think this is necessary here. Honestly, we could get away without any check; I added this as a slight precaution in case a developer in the future changes the code in a breaking way without realizing it immediately.

Reply:
Also to address the duck typing @bdice, that would need to go into the Python box class. We need the C++ BoxDim object. A `from_box` method for `hoomd.box.Box` would be useful though, feel free to raise an issue.

Reply:
Agreed, all such syntactic sugar should be done in pure Python. The C++ API should be type-safe and as type-specific as possible. I would honestly vote to remove this, pybind throws pretty readable errors when developer error like this pops up IMO.